### PR TITLE
skip tests for the wheels. This saves 50% of the size of the wheel

### PR DESCRIPTION
**Pull request by tardyp**
Signed-off-by: Pierre Tardy <pierre.tardy@intel.com>

### CODE REVIEW

File: master/setup.py [hunk 1]
```diff
@@ -166,16 +170,17 @@ def define_plugin_entries(groups):
         "buildbot.steps.package.deb",
         "buildbot.steps.package.rpm",
         "buildbot.steps.source",
-        "buildbot.test",
+        "buildbot.test.util",
```
Comment (sa2ajj) [inline master/setup.py]: If tests are not included, why do we want to still include `buildbot.test.{util,fake}`?

**Comment by sa2ajj** [comment]
Beside the clarification I asked for, LGTM.

File: master/setup.py (hunk 1 above)
Comment (tardyp) [inline master/setup.py]: Because they are part of the framework. They can be used to write tests for other projects using buildbot (e.g buildbot_travis)

**Comment by tardyp** [comment]
added clarification

**Comment by sa2ajj** [comment]
While I agree this is a strong use case, I do not think we have enough documentation for things provided by those modules...

All in all, I'd like to hear at least one more opinion on this - @jaredgrubb, @djmitche, @tomprince?

**Comment by tardyp** [comment]
indeed, we only have those http://docs.buildbot.net/latest/developer/tests.html

But imho, the test system is very well designed and strong. my team would have saved some time if they'd use it instead of rewritting their own.

File: master/setup.py [hunk 2]
```diff
@@ -29,6 +29,10 @@
 from distutils.command.install_data import install_data
 from distutils.command.sdist import sdist
 
+if "bdist_wheel" in sys.argv:
+    NOT_WHEEL = False
+else:
+    NOT_WHEEL = True
```
Comment (djmitche) [inline master/setup.py]: A boolean with "not" in the name is confusing!  Can you reverse this to "BUILDING_WHEEL" or something similar?

**Comment by djmitche** [comment]
I think this is good.  I don't understand your concern, @sa2ajj -- the test utils are, I'll admit, sparsely documented, but that's isolated to the `buildbot.tests` package hierarchy and seems more-or-less par for the course for tests.

**Comment by sa2ajj** [comment]
@djmitche the intention for leaving `buildbot.test.{util,fake}` in is to allow third-party packages to perform tests: since they are 3rd party, they'd need to know how to use stuff provided by these test packages, and we do not have elaborate docs for that.  That's my concern.

**Comment by djmitche** [comment]
Ah, I see.  And I don't like that :)

I consider all of those to be private and only for use within Buildbot -- so if I refactor FakeFoo, I can look for all occurrences of FakeFoo and update them.

Buildbot has historically advertised a very broad and badly-controlled API surface, and that's a big part of the pain of upgrades and refactors.  I don't want to repeat that.

Even the fake DB and Data API components should, I think, be kept private.  While they mostly implement the well-defined and controlled DB and Data APIs, they also have interfaces for adding and examining test data which may change from version to version.  Sometime in the midst of 0.9.x for x > 0 I might be comfortable regularizing and documenting those interfaces, but at the moment they're still pretty fluid.

**Comment by tardyp** [comment]
my opinion is that it is always better to use unsupported api rather than rewriting its own unit test framework.

Buildbot expectCommand framework is awesome for testing custom steps. Even if this is not supported, I would recommend everybody to use that rather than anything in-house.

I would also recommend to use the test.util.integration in order to create integration tests for custom steps.

Anyway If people want this they can still take the tarball.

switched to ternary if, and removed test.util and test.fake

**Comment by sa2ajj** [comment]
LGTM

**Comment by djmitche** [comment]
Let's get a 1.0.0 bug filed to make those test suites available later.

**Comment by tardyp** [comment]
rebased

**Pull request merged**